src/PerformanceSensitiveAnalyzers/UnitTests/AvoidAllocationWithArrayEmptyCodeFixTests.cs

```diff
@@ -0,0 +1,550 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers;
+using Xunit;
+using VerifyCS = Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers.UnitTests.CSharpPerformanceCodeFixVerifier<
+    Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers.ExplicitAllocationAnalyzer,
+    Microsoft.CodeAnalysis.CSharp.PerformanceSensitiveAnalyzers.AvoidAllocationWithArrayEmptyCodeFix>;
+
+namespace Microsoft.CodeAnalysis.PerformanceSensitive.Analyzers.UnitTests
+{
+    public class AvoidAllocationWithArrayEmptyCodeFixTests
+    {
+        public static readonly TheoryData<string> CollectionTypes = new TheoryData<string>
+        {
+            "IEnumerable<int>",
+            "IReadOnlyList<int>",
+            "IReadOnlyCollection<int>"
+        };
+
+        [Theory]
+        [InlineData("IEnumerable<int>")]
+        [InlineData("IReadOnlyList<int>")]
+        [InlineData("IReadOnlyCollection<int>")]
+        public async Task ShouldReplaceEmptyListCreationWithArrayEmptyWithReturnType(string type)
+        {
+            var initial = $@"
+using System;
+using System.Collections.Generic;
+using Roslyn.Utilities;
+
+namespace SampleNamespace
+{{
+    class SampleClass
+    {{
+        [PerformanceSensitive(""uri"")]
+        public {type} DoSomething()
+        {{
+            return new List<int>();
+        }}
+    }}
+}}";
+
+            var expected = $@"
+using System;
+using System.Collections.Generic;
+using Roslyn.Utilities;
+
+namespace SampleNamespace
+{{
+    class SampleClass
+    {{
+        [PerformanceSensitive(""uri"")]
+        public {type} DoSomething()
+        {{
+            return Array.Empty<int>();
```

Comment:
`Enumerable<T>` has two different behaviours, one for optimizing [size](https://github.com/dotnet/runtime/blob/e2dafdfcb3500e1305e26f90d89d66fa35280426/src/libraries/System.Linq/src/System/Linq/Enumerable.SizeOpt.cs#L11) and one for [speed](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Linq/src/System/Linq/Enumerable.SpeedOpt.cs#L11).

Most of the other speed optimizations also return an [empty partition](https://source.dot.net/#System.Linq/System/Linq/Partition.SpeedOpt.cs,9611940f9dbc593c,references). Maybe return `Enumerable.Empty<T>` for the scenarios where there's an `IEnumerable<T>` return type?

Reply:
Thanks. That makes sense. I will make a follow up PR for this, since this is just to port the codefix from its original repo, I'd like to avoid making changes to the logic.

BTW, the behavior of `Enumerable.Empty` in .NET Core is explained in the commit message here: 
https://github.com/dotnet/runtime/commit/311c3a096aea7a6b6bc038d6c1ee334c5a4855b6